modules/dnn/src/webgpu/include/tensor.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef OPENCV_DNN_WEBGPU_TENSOR_HPP
+#define OPENCV_DNN_WEBGPU_TENSOR_HPP
+// #ifdef HAVE_WEBGPU
+#include<dawn/webgpu_cpp.h>
+// #endif
+#include"buffer.hpp"
+#include"wgpucom.hpp"
+#include<memory>
+namespace cv { namespace dnn { namespace webgpu {
+// #ifdef HAVE_WEBGPU
+class Buffer;
+class Tensor{
+public:
+    Tensor(Format fmt = wFormatFp32);
+    Tensor( const void* data, size_t size_in_byte, 
+            wgpu::BufferUsage usage, 
+            Format fmt = wFormatInt32);  //uniform buffer
+    Tensor( const void* data, std::vector<int>& shape, 
+            wgpu::BufferUsage usage, 
+            Format fmt = wFormatFp32);
+    const void* map();
+    void unMap();
+    Shape getShape() const;
+    int dimSize(const int dim) const;
+    int dimNum() const;
+    int count(const int start_axis = 0, const int end_axis = -1) const;
+    // Change shape and format to as passed in.
+    // Copy data if data != NULL
+    // Allocate new internal buffer if new size > old size or alloc flag is true
+    Tensor reshape( const void* data, const std::vector<int>& shape, 
+                    bool alloc = false, 
+                    Format fmt = wFormatInvalid);
+    Tensor setUniform(const void * data, Format fmt = wFormatInt32);
```

Comment:
As my understanding, setUniform is to fill the tensor with cpu accessiable data, right? 
fmt parameter is not necessary here. I suggest the the API as “Tensor fillData(const void *data, const size_t size)” to pair with mapRead().